python/ipywidgets/ipywidgets/widgets/widget.py

```diff
@@ -272,22 +283,46 @@ def register(widget):
     return widget
 
 
+class _staticproperty(object):
+    def __init__(self, fget):
+        self.fget = fget
+
+    def __get__(self, owner_self, owner_cls):
+        assert owner_self is None
+        return self.fget()
+
+
+
 class Widget(LoggingHasTraits):
     #-------------------------------------------------------------------------
     # Class attributes
     #-------------------------------------------------------------------------
     _widget_construction_callback = None
     _control_comm = None
 
-    # _active_widgets is a dictionary of all active widget objects
-    _active_widgets = {}
+    @_staticproperty
+    def widgets():
+        warnings.warn("Widget.widgets is deprecated, use ipywidgets.widgets.widget.instances", DeprecationWarning)
```

Comment:
See https://github.com/jupyter-widgets/ipywidgets/pull/3569 for some utility functions to give deprecation warnings the user will actually see.

Reply:
should we see which PR merges first, if this comes first, we can add changes to #3569 to use that